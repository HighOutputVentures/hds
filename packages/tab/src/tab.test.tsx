import { TabPanel, TabPanels } from '@chakra-ui/react';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Tab from './tab';

describe('Should render Tab', () => {
  beforeEach(() => {
    render(
      <Tab
        tabLabel={['My details', 'Profile', 'Password', 'Team']}
        tabItems={
          <TabPanels>
            <TabPanel>One</TabPanel>
            <TabPanel>Two</TabPanel>
            <TabPanel>Three</TabPanel>
            <TabPanel>Four</TabPanel>
          </TabPanels>
        }
      />
    );
  });

  it('Should render tab labels', async () => {
    const tabs = await screen.findAllByRole('tab');
    expect(tabs).toHaveLength(4);
  });

  it('Should render the current selected tab', () => {
    const tab = screen.getByRole('tab', { name: /My details/i });
    expect(tab).toHaveAttribute('aria-selected', 'true');
  });
});

describe('Snapshot', () => {
  const component = renderer.create(
    <Tab
      tabLabel={['My details', 'Profile', 'Password', 'Team']}
      tabItems={
        <TabPanels>
          <TabPanel>One</TabPanel>
          <TabPanel>Two</TabPanel>
          <TabPanel>Three</TabPanel>
          <TabPanel>Four</TabPanel>
        </TabPanels>
      }
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  component.unmount();
});
