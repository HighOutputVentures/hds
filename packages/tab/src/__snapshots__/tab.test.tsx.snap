// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="css-1bshtmn"
>
  <div
    className="chakra-tabs css-fakzyp"
    data-testid="hds.tabs"
  >
    <div
      aria-orientation="horizontal"
      className="chakra-tabs__tablist css-16mo8a4"
      data-testid="hds.tab.list"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-controls="tabs-:r0:--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-1c44rwm"
        data-testid="hds.tab"
        id="tabs-:r0:--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <p
          className="chakra-text css-0"
        >
          My details
        </p>
      </button>
      <button
        aria-controls="tabs-:r0:--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-1c44rwm"
        data-testid="hds.tab"
        id="tabs-:r0:--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <p
          className="chakra-text css-0"
        >
          Profile
        </p>
      </button>
      <button
        aria-controls="tabs-:r0:--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-1c44rwm"
        data-testid="hds.tab"
        id="tabs-:r0:--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <p
          className="chakra-text css-0"
        >
          Password
        </p>
      </button>
      <button
        aria-controls="tabs-:r0:--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-1c44rwm"
        data-testid="hds.tab"
        id="tabs-:r0:--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <p
          className="chakra-text css-0"
        >
          Team
        </p>
      </button>
    </div>
    <div
      className="chakra-tabs__tab-panels css-j359i1"
      data-testid="hds.tab.panels"
    >
      <div
        className="chakra-tabs__tab-panels css-8atqhb"
      >
        <div
          aria-labelledby="tabs-:r0:--tab-0"
          className="chakra-tabs__tab-panel css-sjt5nk"
          hidden={false}
          id="tabs-:r0:--tabpanel-0"
          role="tabpanel"
          tabIndex={0}
        >
          One
        </div>
        <div
          aria-labelledby="tabs-:r0:--tab-1"
          className="chakra-tabs__tab-panel css-sjt5nk"
          hidden={true}
          id="tabs-:r0:--tabpanel-1"
          role="tabpanel"
          tabIndex={0}
        >
          Two
        </div>
        <div
          aria-labelledby="tabs-:r0:--tab-2"
          className="chakra-tabs__tab-panel css-sjt5nk"
          hidden={true}
          id="tabs-:r0:--tabpanel-2"
          role="tabpanel"
          tabIndex={0}
        >
          Three
        </div>
        <div
          aria-labelledby="tabs-:r0:--tab-3"
          className="chakra-tabs__tab-panel css-sjt5nk"
          hidden={true}
          id="tabs-:r0:--tabpanel-3"
          role="tabpanel"
          tabIndex={0}
        >
          Four
        </div>
      </div>
    </div>
  </div>
</div>
`;
